+++ src/app/numptyphysics/Canvas.cpp
@@ -28,9 +28,11 @@
 #include "Swipe.h"
 #include <SDL/SDL_syswm.h>
 #ifndef WIN32
+#ifndef GENODE
 #include <X11/X.h>
 #include <X11/Xlib.h>
 #endif
+#endif
 #undef Window
 
 //#define FORCE_16BPP
@@ -759,7 +761,7 @@ void Window::raise()
   SDL_VERSION( &sys.version );
   SDL_GetWMInfo( &sys );
 
-#if !defined(WIN32) && !(defined(__APPLE__) && defined(__MACH__))
+#if !defined(WIN32) && !defined(GENODE) && !(defined(__APPLE__) && defined(__MACH__))
   /* No X11 stuff on Windows and Mac OS X */
 
   // take focus...
