--- src/app/cmus/server.c
+++ src/app/cmus/server.c
@@ -51,7 +51,7 @@
 #include <fcntl.h>
 #include <ctype.h>
 
-int server_socket;
+int server_socket = -1;
 LIST_HEAD(client_head);
 
 static char *title_buf = NULL;
@@ -365,7 +365,7 @@ void server_init(char *address)
 
 	server_socket = socket(addr.sa.sa_family, SOCK_STREAM, 0);
 	if (server_socket == -1)
-		die_errno("socket");
+		return;
 
 	if (bind(server_socket, &addr.sa, addrlen) == -1) {
 		int sock;
--- src/app/cmus/ui_curses.c
+++ src/app/cmus/ui_curses.c
@@ -2185,7 +2185,8 @@ static void main_loop(void)
 		SELECT_ADD_FD(0);
 		SELECT_ADD_FD(job_fd);
 		SELECT_ADD_FD(cmus_next_track_request_fd);
-		SELECT_ADD_FD(server_socket);
+		if (server_socket > 0)
+			SELECT_ADD_FD(server_socket);
 		if (mpris_fd != -1)
 			SELECT_ADD_FD(mpris_fd);
 		list_for_each_entry(client, &client_head, node) {
@@ -2233,7 +2234,7 @@ static void main_loop(void)
 				update_statusline();
 			}
 		}
-		if (FD_ISSET(server_socket, &set))
+		if (server_socket > 0 && FD_ISSET(server_socket, &set))
 			server_accept();
 
 		// server_serve() can remove client from the list
