chromium_flags.patch

diff --git a/src/lib/app/mainapplication.cpp b/src/lib/app/mainapplication.cpp
index a769318cb..959d60a68 100644
--- a/src/lib/app/mainapplication.cpp
+++ b/src/lib/app/mainapplication.cpp
@@ -153,7 +153,13 @@ MainApplication::MainApplication(int &argc, char** argv)
 
     QByteArray chromium_flags = qgetenv("QTWEBENGINE_CHROMIUM_FLAGS");
     chromium_flags.append(" --enable-features=WebRTCPipeWireCapturer");
+#if 0
+	/*
+	 * Setting this environment variable apparently disabled the handling
+	 * of command line flags.
+	 */
     qputenv("QTWEBENGINE_CHROMIUM_FLAGS", chromium_flags);
+#endif
 
     QUrl startUrl;
     QString startProfile;
