genode_no_qprocess.patch

diff --git a/src/lib/app/mainapplication.cpp b/src/lib/app/mainapplication.cpp
index 88d78709b..9837d5939 100644
--- a/src/lib/app/mainapplication.cpp
+++ b/src/lib/app/mainapplication.cpp
@@ -258,7 +258,7 @@ MainApplication::MainApplication(int &argc, char** argv)
     if (messages.isEmpty()) {
         messages.append(QStringLiteral(" "));
     }
-
+#if 0
     if (isRunning()) {
         m_isClosing = true;
         for (const QString &message : qAsConst(messages)) {
@@ -266,6 +266,7 @@ MainApplication::MainApplication(int &argc, char** argv)
         }
         return;
     }
+#endif
 
 #ifdef Q_OS_MACOS
     setQuitOnLastWindowClosed(false);
@@ -694,9 +695,9 @@ void MainApplication::startPrivateBrowsing(const QUrl &startUrl)
         args << url.toEncoded();
     }
 
-    if (!QProcess::startDetached(applicationFilePath(), args)) {
+    //if (!QProcess::startDetached(applicationFilePath(), args)) {
         qWarning() << "MainApplication: Cannot start new browser process for private browsing!" << applicationFilePath() << args;
-    }
+    //}
 }
 
 void MainApplication::reloadUserStyleSheet()
diff --git a/src/lib/tools/qztools.cpp b/src/lib/tools/qztools.cpp
index fbdad97da..e05c2fc23 100644
--- a/src/lib/tools/qztools.cpp
+++ b/src/lib/tools/qztools.cpp
@@ -847,6 +847,7 @@ QStringList QzTools::splitCommandArguments(const QString &command)
 
 bool QzTools::startExternalProcess(const QString &executable, const QString &args)
 {
+#if 0
     const QStringList arguments = splitCommandArguments(args);
 
     bool success = QProcess::startDetached(executable, arguments);
@@ -861,6 +862,9 @@ bool QzTools::startExternalProcess(const QString &executable, const QString &arg
     }
 
     return success;
+#else
+	return false;
+#endif
 }
 
 void QzTools::setWmClass(const QString &name, const QWidget* widget)
