no_ecm.patch

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 2ed63ab07..5f591f370 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -12,20 +12,20 @@ project(Falkon VERSION ${RELEASE_SERVICE_VERSION})
 
 # Find ECM, with nice error handling in case of failure
 include(FeatureSummary)
-find_package(ECM 5.78.0 CONFIG)
-set_package_properties(ECM PROPERTIES TYPE REQUIRED DESCRIPTION "Extra CMake Modules." URL "https://invent.kde.org/frameworks/extra-cmake-modules")
+#find_package(ECM 5.78.0 CONFIG)
+#set_package_properties(ECM PROPERTIES TYPE REQUIRED DESCRIPTION "Extra CMake Modules." URL "https://invent.kde.org/frameworks/extra-cmake-modules")
 feature_summary(WHAT REQUIRED_PACKAGES_NOT_FOUND FATAL_ON_MISSING_REQUIRED_PACKAGES)
 set(CMAKE_MODULE_PATH ${ECM_MODULE_PATH} ${CMAKE_CURRENT_SOURCE_DIR}/cmake)
 
 # Many includes from ECM, to get all the nice cmake functions and settings
-include(KDEInstallDirs)
-include(KDECMakeSettings)
-include(KDECompilerSettings NO_POLICY_SCOPE)
-include(ECMInstallIcons)
-include(ECMSetupVersion)
-include(ECMAddAppIcon)
-include(ECMQtDeclareLoggingCategory)
-include(ECMPoQmTools)
+#include(KDEInstallDirs)
+#include(KDECMakeSettings)
+#include(KDECompilerSettings NO_POLICY_SCOPE)
+#include(ECMInstallIcons)
+#include(ECMSetupVersion)
+#include(ECMAddAppIcon)
+#include(ECMQtDeclareLoggingCategory)
+#include(ECMPoQmTools)
 if (QT_MAJOR_VERSION STREQUAL "6")
     set(QT_MIN_VERSION "6.4.0")
     set(KF_MIN_VERSION "5.240.0")
diff --git a/src/lib/CMakeLists.txt b/src/lib/CMakeLists.txt
index 764701ff0..29173103a 100644
--- a/src/lib/CMakeLists.txt
+++ b/src/lib/CMakeLists.txt
@@ -501,7 +501,7 @@ set(SRCS ${SRCS}
 )
 
 # add the logging category for FALKON
-ecm_qt_declare_logging_category(LOGGING_SRCS HEADER falkon_private_debug.h IDENTIFIER "FALKON_PRIVATE_LOG" CATEGORY_NAME "org.kde.falkon")
+#ecm_qt_declare_logging_category(LOGGING_SRCS HEADER falkon_private_debug.h IDENTIFIER "FALKON_PRIVATE_LOG" CATEGORY_NAME "org.kde.falkon")
 
 if (HAVE_LIBINTL)
     set(SRCS ${SRCS} plugins/qml/api/i18n/qmli18n.cpp plugins/qml/api/i18n/qmli18n.h)
diff --git a/src/main/CMakeLists.txt b/src/main/CMakeLists.txt
index b5ad441f9..5ebc06322 100644
--- a/src/main/CMakeLists.txt
+++ b/src/main/CMakeLists.txt
@@ -10,7 +10,7 @@ if (WIN32)
     set(SRCS ${SRCS} version.rc)
     configure_file(falkonversion.h.cmake ${CMAKE_CURRENT_BINARY_DIR}/falkonversion.h)
 endif()
-ecm_create_qm_loader(SRCS falkon_qt)
+#ecm_create_qm_loader(SRCS falkon_qt)
 add_executable(falkon ${SRCS})
 
 target_link_libraries(falkon FalkonPrivate)
@@ -35,21 +35,21 @@ if (UNIX AND NOT APPLE)
     if (KF${KF_MAJOR_VERSION}I18n_FOUND)
         ki18n_install(${CMAKE_SOURCE_DIR}/po)
     endif()
-    ecm_install_po_files_as_qm(${CMAKE_SOURCE_DIR}/poqm)
-
-    install(PROGRAMS ../../linux/applications/org.kde.falkon.desktop DESTINATION ${KDE_INSTALL_APPDIR})
-
-    ecm_install_icons(ICONS
-        ../../linux/hicolor/16-apps-falkon.png
-        ../../linux/hicolor/32-apps-falkon.png
-        ../../linux/hicolor/48-apps-falkon.png
-        ../../linux/hicolor/64-apps-falkon.png
-        ../../linux/hicolor/128-apps-falkon.png
-        ../../linux/hicolor/256-apps-falkon.png
-        ../../linux/hicolor/sc-apps-falkon.svg
-        DESTINATION ${KDE_INSTALL_ICONDIR} THEME hicolor)
-
-    install(FILES ../../linux/completion/falkon DESTINATION ${KDE_INSTALL_DATADIR}/bash-completion/completions)
-    install(FILES ../../linux/appdata/org.kde.falkon.appdata.xml DESTINATION ${KDE_INSTALL_METAINFODIR})
+#    ecm_install_po_files_as_qm(${CMAKE_SOURCE_DIR}/poqm)
+
+#    install(PROGRAMS ../../linux/applications/org.kde.falkon.desktop DESTINATION ${KDE_INSTALL_APPDIR})
+
+#    ecm_install_icons(ICONS
+#        ../../linux/hicolor/16-apps-falkon.png
+#        ../../linux/hicolor/32-apps-falkon.png
+#        ../../linux/hicolor/48-apps-falkon.png
+#        ../../linux/hicolor/64-apps-falkon.png
+#        ../../linux/hicolor/128-apps-falkon.png
+#        ../../linux/hicolor/256-apps-falkon.png
+#        ../../linux/hicolor/sc-apps-falkon.svg
+#        DESTINATION ${KDE_INSTALL_ICONDIR} THEME hicolor)
+
+#    install(FILES ../../linux/completion/falkon DESTINATION ${KDE_INSTALL_DATADIR}/bash-completion/completions)
+#    install(FILES ../../linux/appdata/org.kde.falkon.appdata.xml DESTINATION ${KDE_INSTALL_METAINFODIR})
 endif()
 
