--- src/hogweed.pc.in.orig	2025-03-03 18:24:50.114205622 +0100
+++ src/hogweed.pc.in	2025-03-03 18:24:37.060727286 +0100
@@ -1,8 +1,3 @@
-prefix=@prefix@
-exec_prefix=@exec_prefix@
-libdir=@libdir@
-includedir=@includedir@
-
 # Uses Requires.private and Libs.private, under the assumption that
 # when using shared libraries, the ELF dependencies from libhogweed.so
 # to nettle and gmp work.
@@ -11,9 +6,4 @@ Name: Hogweed
 Description: Nettle low-level cryptographic library (public-key algorithms)
 URL: http://www.lysator.liu.se/~nisse/nettle
 Version: @PACKAGE_VERSION@
-Requires: @IF_NOT_SHARED@ nettle
-Requires.private: @IF_SHARED@ nettle
-Libs: -L${libdir} -lhogweed @IF_NOT_SHARED@ @LIBS@
-Libs.private: @IF_SHARED@ @LIBS@
-Cflags: -I${includedir}
-
+Libs: 
