diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/golang.org/x/net/route/interface_classic.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/golang.org/x/net/route/interface_classic.go
--- contrib/gcc.libgo_etalon/libgo/go/golang.org/x/net/route/interface_classic.go	2023-06-01 13:13:31.202417073 +0300
+++ gcc/libgo/go/golang.org/x/net/route/interface_classic.go	2023-06-01 13:31:19.752552073 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build darwin || dragonfly || netbsd
+//go:build darwin || dragonfly || netbsd || genode
 // +build darwin dragonfly netbsd
 
 package route
@@ -54,6 +54,8 @@
 	}
 	if runtime.GOOS == "netbsd" {
 		m.Index = int(nativeEndian.Uint16(b[16:18]))
+	} else if runtime.GOOS == "genode" {
+		m.Index = int(nativeEndian.Uint16(b[16:18]))
 	} else {
 		m.Index = int(nativeEndian.Uint16(b[12:14]))
 	}
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/golang.org/x/net/route/route.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/golang.org/x/net/route/route.go
--- contrib/gcc.libgo_etalon/libgo/go/golang.org/x/net/route/route.go	2023-06-01 13:13:31.203417086 +0300
+++ gcc/libgo/go/golang.org/x/net/route/route.go	2023-06-01 13:32:24.700409223 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build darwin || dragonfly || freebsd || netbsd || openbsd
+//go:build darwin || dragonfly || freebsd || netbsd || openbsd || genode
 // +build darwin dragonfly freebsd netbsd openbsd
 
 // Package route provides basic functions for the manipulation of
@@ -92,8 +92,8 @@
 type RIBType int
 
 const (
-	RIBTypeRoute     RIBType = syscall.NET_RT_DUMP
-	RIBTypeInterface RIBType = syscall.NET_RT_IFLIST
+	RIBTypeRoute     RIBType = syscall._NET_RT_DUMP
+	RIBTypeInterface RIBType = syscall._NET_RT_IFLIST
 )
 
 // FetchRIB fetches a routing information base from the operating

