diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/golang.org/x/net/route/syscall.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/golang.org/x/net/route/syscall.go
--- contrib/gcc.libgo_etalon/libgo/go/golang.org/x/net/route/syscall.go	2023-06-01 13:13:31.202417073 +0300
+++ gcc/libgo/go/golang.org/x/net/route/syscall.go	2023-06-01 13:33:09.276996563 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build darwin || dragonfly || freebsd || netbsd || openbsd
+//go:build darwin || dragonfly || freebsd || netbsd || openbsd || genode
 // +build darwin dragonfly freebsd netbsd openbsd
 
 package route
