diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/archive/tar/stat_actime1.go gcc/libgo/go/archive/tar/stat_actime1.go
--- contrib/gcc.libgo_etalon/libgo/go/archive/tar/stat_actime1.go	2023-06-01 13:13:31.319418637 +0300
+++ gcc/libgo/go/archive/tar/stat_actime1.go	2023-06-01 13:24:59.273517833 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || hurd || linux || dragonfly || openbsd || solaris
+//go:build aix || hurd || linux || dragonfly || openbsd || solaris || genode
 
 package tar
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/cmd/go/internal/base/signal_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/cmd/go/internal/base/signal_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/cmd/go/internal/base/signal_unix.go	2023-06-01 13:13:31.558421834 +0300
+++ gcc/libgo/go/cmd/go/internal/base/signal_unix.go	2023-06-01 13:26:44.840914244 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || js || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || js || linux || netbsd || openbsd || solaris || genode
 
 package base
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/cmd/go/internal/modload/stat_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/cmd/go/internal/modload/stat_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/cmd/go/internal/modload/stat_unix.go	2023-06-01 13:13:31.555421793 +0300
+++ gcc/libgo/go/cmd/go/internal/modload/stat_unix.go	2023-06-01 13:27:51.597797437 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || genode
 
 package modload
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/crypto/x509/root_bsd.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/crypto/x509/root_bsd.go
--- contrib/gcc.libgo_etalon/libgo/go/crypto/x509/root_bsd.go	2023-06-01 13:13:31.368419293 +0300
+++ gcc/libgo/go/crypto/x509/root_bsd.go	2023-06-01 13:28:47.882542176 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build dragonfly || freebsd || netbsd || openbsd
+//go:build dragonfly || freebsd || netbsd || openbsd || genode
 
 package x509
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/crypto/x509/root_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/crypto/x509/root_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/crypto/x509/root_unix.go	2023-06-01 13:13:31.369419306 +0300
+++ gcc/libgo/go/crypto/x509/root_unix.go	2023-06-01 13:29:11.061848900 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris
+//go:build aix || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || genode
 
 package x509
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/golang.org/x/net/route/address.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/golang.org/x/net/route/address.go
--- contrib/gcc.libgo_etalon/libgo/go/golang.org/x/net/route/address.go	2023-06-01 13:13:31.202417073 +0300
+++ gcc/libgo/go/golang.org/x/net/route/address.go	2023-06-01 13:30:10.602636847 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build darwin || dragonfly || freebsd || netbsd || openbsd
+//go:build darwin || dragonfly || freebsd || netbsd || openbsd || genode
 // +build darwin dragonfly freebsd netbsd openbsd
 
 package route
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/golang.org/x/net/route/binary.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/golang.org/x/net/route/binary.go
--- contrib/gcc.libgo_etalon/libgo/go/golang.org/x/net/route/binary.go	2023-06-01 13:13:31.204417100 +0300
+++ gcc/libgo/go/golang.org/x/net/route/binary.go	2023-06-01 13:30:32.075921041 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build darwin || dragonfly || freebsd || netbsd || openbsd
+//go:build darwin || dragonfly || freebsd || netbsd || openbsd || genode
 // +build darwin dragonfly freebsd netbsd openbsd
 
 package route
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/golang.org/x/net/route/interface.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/golang.org/x/net/route/interface.go
--- contrib/gcc.libgo_etalon/libgo/go/golang.org/x/net/route/interface.go	2023-06-01 13:13:31.203417086 +0300
+++ gcc/libgo/go/golang.org/x/net/route/interface.go	2023-06-01 13:31:32.280717901 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build darwin || dragonfly || freebsd || netbsd || openbsd
+//go:build darwin || dragonfly || freebsd || netbsd || openbsd || genode
 // +build darwin dragonfly freebsd netbsd openbsd
 
 package route
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/golang.org/x/net/route/interface_announce.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/golang.org/x/net/route/interface_announce.go
--- contrib/gcc.libgo_etalon/libgo/go/golang.org/x/net/route/interface_announce.go	2023-06-01 13:13:31.203417086 +0300
+++ gcc/libgo/go/golang.org/x/net/route/interface_announce.go	2023-06-01 13:30:59.918289546 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build dragonfly || freebsd || netbsd
+//go:build dragonfly || freebsd || netbsd || genode
 // +build dragonfly freebsd netbsd
 
 package route
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/golang.org/x/net/route/message.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/golang.org/x/net/route/message.go
--- contrib/gcc.libgo_etalon/libgo/go/golang.org/x/net/route/message.go	2023-06-01 13:13:31.203417086 +0300
+++ gcc/libgo/go/golang.org/x/net/route/message.go	2023-06-01 13:31:57.192046813 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build darwin || dragonfly || freebsd || netbsd || openbsd
+//go:build darwin || dragonfly || freebsd || netbsd || openbsd || genode
 // +build darwin dragonfly freebsd netbsd openbsd
 
 package route
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/golang.org/x/net/route/route_classic.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/golang.org/x/net/route/route_classic.go
--- contrib/gcc.libgo_etalon/libgo/go/golang.org/x/net/route/route_classic.go	2023-06-01 13:13:31.203417086 +0300
+++ gcc/libgo/go/golang.org/x/net/route/route_classic.go	2023-06-01 13:32:12.055242625 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build darwin || dragonfly || freebsd || netbsd
+//go:build darwin || dragonfly || freebsd || netbsd || genode
 // +build darwin dragonfly freebsd netbsd
 
 package route
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/golang.org/x/net/route/sys.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/golang.org/x/net/route/sys.go
--- contrib/gcc.libgo_etalon/libgo/go/golang.org/x/net/route/sys.go	2023-06-01 13:13:31.204417100 +0300
+++ gcc/libgo/go/golang.org/x/net/route/sys.go	2023-06-01 13:32:52.394774114 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build darwin || dragonfly || freebsd || netbsd || openbsd
+//go:build darwin || dragonfly || freebsd || netbsd || openbsd || genode
 // +build darwin dragonfly freebsd netbsd openbsd
 
 package route
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/internal/poll/fcntl_libc.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/internal/poll/fcntl_libc.go
--- contrib/gcc.libgo_etalon/libgo/go/internal/poll/fcntl_libc.go	2023-06-01 13:13:31.146416324 +0300
+++ gcc/libgo/go/internal/poll/fcntl_libc.go	2023-06-01 13:33:42.220430674 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || solaris
+//go:build aix || darwin || solaris || genode
 
 package poll
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/internal/poll/fd_fsync_posix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/internal/poll/fd_fsync_posix.go
--- contrib/gcc.libgo_etalon/libgo/go/internal/poll/fd_fsync_posix.go	2023-06-01 13:13:31.144416297 +0300
+++ gcc/libgo/go/internal/poll/fd_fsync_posix.go	2023-06-01 13:34:06.650752632 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris
+//go:build aix || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || genode
 
 package poll
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/internal/poll/fd_posix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/internal/poll/fd_posix.go
--- contrib/gcc.libgo_etalon/libgo/go/internal/poll/fd_posix.go	2023-06-01 13:13:31.145416311 +0300
+++ gcc/libgo/go/internal/poll/fd_posix.go	2023-06-01 13:34:30.516067165 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows || genode
 
 package poll
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/internal/poll/fd_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/internal/poll/fd_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/internal/poll/fd_unix.go	2023-06-01 13:13:31.146416324 +0300
+++ gcc/libgo/go/internal/poll/fd_unix.go	2023-06-01 13:34:45.532265083 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || genode
 
 package poll
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/internal/poll/hook_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/internal/poll/hook_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/internal/poll/hook_unix.go	2023-06-01 13:13:31.145416311 +0300
+++ gcc/libgo/go/internal/poll/hook_unix.go	2023-06-01 13:35:01.858480276 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || genode
 
 package poll
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/internal/poll/sockopt.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/internal/poll/sockopt.go
--- contrib/gcc.libgo_etalon/libgo/go/internal/poll/sockopt.go	2023-06-01 13:13:31.145416311 +0300
+++ gcc/libgo/go/internal/poll/sockopt.go	2023-06-01 13:35:40.192985598 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || windows
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || windows || genode
 
 package poll
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/internal/poll/sockopt_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/internal/poll/sockopt_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/internal/poll/sockopt_unix.go	2023-06-01 13:13:31.147416337 +0300
+++ gcc/libgo/go/internal/poll/sockopt_unix.go	2023-06-01 13:35:20.247722674 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || genode
 
 package poll
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/internal/poll/sockoptip.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/internal/poll/sockoptip.go
--- contrib/gcc.libgo_etalon/libgo/go/internal/poll/sockoptip.go	2023-06-01 13:13:31.148416351 +0300
+++ gcc/libgo/go/internal/poll/sockoptip.go	2023-06-01 13:36:01.006259980 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || windows
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || windows || genode
 
 package poll
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/internal/poll/sys_cloexec.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/internal/poll/sys_cloexec.go
--- contrib/gcc.libgo_etalon/libgo/go/internal/poll/sys_cloexec.go	2023-06-01 13:13:31.147416337 +0300
+++ gcc/libgo/go/internal/poll/sys_cloexec.go	2023-06-01 13:36:20.824521259 +0300
@@ -5,7 +5,7 @@
 // This file implements accept for platforms that do not provide a fast path for
 // setting SetNonblock and CloseOnExec.
 
-//go:build aix || darwin || (js && wasm) || (solaris && !illumos)
+//go:build aix || darwin || (js && wasm) || (solaris && !illumos) || genode
 
 package poll
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/internal/syscall/unix/nonblocking_libc.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/internal/syscall/unix/nonblocking_libc.go
--- contrib/gcc.libgo_etalon/libgo/go/internal/syscall/unix/nonblocking_libc.go	2023-06-01 13:13:31.130416110 +0300
+++ gcc/libgo/go/internal/syscall/unix/nonblocking_libc.go	2023-06-01 13:38:01.126843846 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || solaris
+//go:build aix || darwin || solaris || genode
 
 package unix
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/addrselect.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/addrselect.go
--- contrib/gcc.libgo_etalon/libgo/go/net/addrselect.go	2023-06-01 13:13:31.350419052 +0300
+++ gcc/libgo/go/net/addrselect.go	2023-06-01 13:38:40.304360540 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || genode
 
 // Minimal RFC 6724 address selection.
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/conf.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/conf.go
--- contrib/gcc.libgo_etalon/libgo/go/net/conf.go	2023-06-01 13:13:31.332418811 +0300
+++ gcc/libgo/go/net/conf.go	2023-06-01 13:41:07.804306342 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/dnsclient_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/dnsclient_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/net/dnsclient_unix.go	2023-06-01 13:13:31.331418798 +0300
+++ gcc/libgo/go/net/dnsclient_unix.go	2023-06-01 13:41:33.684647831 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || genode
 
 // DNS client: see RFC 1035.
 // Has to be linked into package net for Dial.
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/dnsconfig_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/dnsconfig_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/net/dnsconfig_unix.go	2023-06-01 13:13:31.327418744 +0300
+++ gcc/libgo/go/net/dnsconfig_unix.go	2023-06-01 13:41:53.676911643 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || genode
 
 // Read system DNS config from /etc/resolv.conf
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/error_posix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/error_posix.go
--- contrib/gcc.libgo_etalon/libgo/go/net/error_posix.go	2023-06-01 13:13:31.350419052 +0300
+++ gcc/libgo/go/net/error_posix.go	2023-06-01 13:42:14.354184509 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/error_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/error_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/net/error_unix.go	2023-06-01 13:13:31.327418744 +0300
+++ gcc/libgo/go/net/error_unix.go	2023-06-01 13:42:30.546398199 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || js || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || js || linux || netbsd || openbsd || solaris || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/fd_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/fd_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/net/fd_unix.go	2023-06-01 13:13:31.346418999 +0300
+++ gcc/libgo/go/net/fd_unix.go	2023-06-01 13:42:46.601610089 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/file_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/file_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/net/file_unix.go	2023-06-01 13:13:31.344418972 +0300
+++ gcc/libgo/go/net/file_unix.go	2023-06-01 13:43:03.992839621 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/hook_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/hook_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/net/hook_unix.go	2023-06-01 13:13:31.333418825 +0300
+++ gcc/libgo/go/net/hook_unix.go	2023-06-01 13:43:22.117078838 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/interface_stub.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/interface_stub.go
--- contrib/gcc.libgo_etalon/libgo/go/net/interface_stub.go	2023-06-01 13:13:31.347419012 +0300
+++ gcc/libgo/go/net/interface_stub.go	2023-06-01 13:43:44.254371038 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build hurd || (js && wasm)
+//go:build hurd || (js && wasm) || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/iprawsock_posix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/iprawsock_posix.go
--- contrib/gcc.libgo_etalon/libgo/go/net/iprawsock_posix.go	2023-06-01 13:13:31.334418838 +0300
+++ gcc/libgo/go/net/iprawsock_posix.go	2023-06-01 13:44:23.795893005 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/ipsock_posix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/ipsock_posix.go
--- contrib/gcc.libgo_etalon/libgo/go/net/ipsock_posix.go	2023-06-01 13:13:31.327418744 +0300
+++ gcc/libgo/go/net/ipsock_posix.go	2023-06-01 13:44:40.024107240 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/lookup_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/lookup_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/net/lookup_unix.go	2023-06-01 13:13:31.347419012 +0300
+++ gcc/libgo/go/net/lookup_unix.go	2023-06-01 13:44:58.951357117 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/nss.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/nss.go
--- contrib/gcc.libgo_etalon/libgo/go/net/nss.go	2023-06-01 13:13:31.347419012 +0300
+++ gcc/libgo/go/net/nss.go	2023-06-01 13:45:15.940581418 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/port_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/port_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/net/port_unix.go	2023-06-01 13:13:31.334418838 +0300
+++ gcc/libgo/go/net/port_unix.go	2023-06-01 13:45:33.135808449 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || genode
 
 // Read system port mappings from /etc/services
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/sendfile_stub.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/sendfile_stub.go
--- contrib/gcc.libgo_etalon/libgo/go/net/sendfile_stub.go	2023-06-01 13:13:31.353419092 +0300
+++ gcc/libgo/go/net/sendfile_stub.go	2023-06-01 13:46:03.458208823 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || (js && wasm) || netbsd || openbsd || ios
+//go:build aix || (js && wasm) || netbsd || openbsd || ios || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/sock_posix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/sock_posix.go
--- contrib/gcc.libgo_etalon/libgo/go/net/sock_posix.go	2023-06-01 13:13:31.332418811 +0300
+++ gcc/libgo/go/net/sock_posix.go	2023-06-01 13:46:25.371498182 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || windows
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || windows || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/sock_stub.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/sock_stub.go
--- contrib/gcc.libgo_etalon/libgo/go/net/sock_stub.go	2023-06-01 13:13:31.347419012 +0300
+++ gcc/libgo/go/net/sock_stub.go	2023-06-01 13:46:37.909663752 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || hurd || (js && wasm) || solaris
+//go:build aix || hurd || (js && wasm) || solaris || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/sockaddr_posix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/sockaddr_posix.go
--- contrib/gcc.libgo_etalon/libgo/go/net/sockaddr_posix.go	2023-06-01 13:13:31.352419079 +0300
+++ gcc/libgo/go/net/sockaddr_posix.go	2023-06-01 13:46:54.012876408 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/sockopt_bsd.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/sockopt_bsd.go
--- contrib/gcc.libgo_etalon/libgo/go/net/sockopt_bsd.go	2023-06-01 13:13:31.348419025 +0300
+++ gcc/libgo/go/net/sockopt_bsd.go	2023-06-01 13:47:10.786097919 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build darwin || dragonfly || freebsd || netbsd || openbsd
+//go:build darwin || dragonfly || freebsd || netbsd || openbsd || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/sockopt_posix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/sockopt_posix.go
--- contrib/gcc.libgo_etalon/libgo/go/net/sockopt_posix.go	2023-06-01 13:13:31.325418718 +0300
+++ gcc/libgo/go/net/sockopt_posix.go	2023-06-01 13:47:29.345343028 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || windows
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || windows || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/sockoptip_bsdvar.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/sockoptip_bsdvar.go
--- contrib/gcc.libgo_etalon/libgo/go/net/sockoptip_bsdvar.go	2023-06-01 13:13:31.348419025 +0300
+++ gcc/libgo/go/net/sockoptip_bsdvar.go	2023-06-01 13:47:46.978575918 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || netbsd || openbsd || solaris || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/sockoptip_posix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/sockoptip_posix.go
--- contrib/gcc.libgo_etalon/libgo/go/net/sockoptip_posix.go	2023-06-01 13:13:31.349419039 +0300
+++ gcc/libgo/go/net/sockoptip_posix.go	2023-06-01 13:48:06.395832381 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || windows
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || windows || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/sys_cloexec.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/sys_cloexec.go
--- contrib/gcc.libgo_etalon/libgo/go/net/sys_cloexec.go	2023-06-01 13:13:31.347419012 +0300
+++ gcc/libgo/go/net/sys_cloexec.go	2023-06-01 13:48:24.683073929 +0300
@@ -5,7 +5,7 @@
 // This file implements sysSocket for platforms that do not provide a fast path
 // for setting SetNonblock and CloseOnExec.
 
-//go:build aix || darwin || (solaris && !illumos)
+//go:build aix || darwin || (solaris && !illumos) || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/tcpsock_posix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/tcpsock_posix.go
--- contrib/gcc.libgo_etalon/libgo/go/net/tcpsock_posix.go	2023-06-01 13:13:31.326418731 +0300
+++ gcc/libgo/go/net/tcpsock_posix.go	2023-06-01 13:48:38.927262082 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/tcpsockopt_posix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/tcpsockopt_posix.go
--- contrib/gcc.libgo_etalon/libgo/go/net/tcpsockopt_posix.go	2023-06-01 13:13:31.345418985 +0300
+++ gcc/libgo/go/net/tcpsockopt_posix.go	2023-06-01 13:48:57.910512844 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || windows
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || windows || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/tcpsockopt_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/tcpsockopt_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/net/tcpsockopt_unix.go	2023-06-01 13:13:31.349419039 +0300
+++ gcc/libgo/go/net/tcpsockopt_unix.go	2023-06-01 13:49:10.997685830 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || freebsd || hurd || linux || netbsd
+//go:build aix || freebsd || hurd || linux || netbsd || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/udpsock_posix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/udpsock_posix.go
--- contrib/gcc.libgo_etalon/libgo/go/net/udpsock_posix.go	2023-06-01 13:13:31.329418771 +0300
+++ gcc/libgo/go/net/udpsock_posix.go	2023-06-01 13:49:25.922883773 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/unixsock_posix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/unixsock_posix.go
--- contrib/gcc.libgo_etalon/libgo/go/net/unixsock_posix.go	2023-06-01 13:13:31.349419039 +0300
+++ gcc/libgo/go/net/unixsock_posix.go	2023-06-01 13:49:39.856068564 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/net/writev_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/net/writev_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/net/writev_unix.go	2023-06-01 13:13:31.327418744 +0300
+++ gcc/libgo/go/net/writev_unix.go	2023-06-01 13:49:59.464328627 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build darwin || dragonfly || freebsd || illumos || linux || netbsd || openbsd
+//go:build darwin || dragonfly || freebsd || illumos || linux || netbsd || openbsd || genode
 
 package net
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/os/error_posix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/os/error_posix.go
--- contrib/gcc.libgo_etalon/libgo/go/os/error_posix.go	2023-06-01 13:13:31.264417902 +0300
+++ gcc/libgo/go/os/error_posix.go	2023-06-01 13:50:26.492687113 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows || genode
 
 package os
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/os/exec/lp_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/os/exec/lp_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/os/exec/lp_unix.go	2023-06-01 13:13:31.260417848 +0300
+++ gcc/libgo/go/os/exec/lp_unix.go	2023-06-01 13:56:51.262791801 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || genode
 
 package exec
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/os/exec_posix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/os/exec_posix.go
--- contrib/gcc.libgo_etalon/libgo/go/os/exec_posix.go	2023-06-01 13:13:31.257417808 +0300
+++ gcc/libgo/go/os/exec_posix.go	2023-06-01 13:50:43.834917135 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows || genode
 
 package os
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/os/exec_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/os/exec_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/os/exec_unix.go	2023-06-01 13:13:31.262417875 +0300
+++ gcc/libgo/go/os/exec_unix.go	2023-06-01 13:50:59.070119215 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || genode
 
 package os
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/os/file_posix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/os/file_posix.go
--- contrib/gcc.libgo_etalon/libgo/go/os/file_posix.go	2023-06-01 13:13:31.254417768 +0300
+++ gcc/libgo/go/os/file_posix.go	2023-06-01 13:52:05.626002061 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows || genode
 
 package os
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/os/file_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/os/file_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/os/file_unix.go	2023-06-01 13:13:31.261417862 +0300
+++ gcc/libgo/go/os/file_unix.go	2023-06-01 13:52:20.741202570 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || genode
 
 package os
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/os/path_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/os/path_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/os/path_unix.go	2023-06-01 13:13:31.264417902 +0300
+++ gcc/libgo/go/os/path_unix.go	2023-06-01 13:52:59.149712092 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || genode
 
 package os
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/os/pipe_bsd.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/os/pipe_bsd.go
--- contrib/gcc.libgo_etalon/libgo/go/os/pipe_bsd.go	2023-06-01 13:13:31.262417875 +0300
+++ gcc/libgo/go/os/pipe_bsd.go	2023-06-01 13:53:20.039989231 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || (js && wasm) || (solaris && !illumos)
+//go:build aix || darwin || (js && wasm) || (solaris && !illumos) || genode
 
 package os
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/os/signal/internal/pty/pty.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/os/signal/internal/pty/pty.go
--- contrib/gcc.libgo_etalon/libgo/go/os/signal/internal/pty/pty.go	2023-06-01 13:13:31.266417929 +0300
+++ gcc/libgo/go/os/signal/internal/pty/pty.go	2023-06-01 13:58:51.489387326 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build (aix || darwin || dragonfly || freebsd || hurd || (linux && !android) || netbsd || openbsd || solaris) && cgo
+//go:build (aix || darwin || dragonfly || freebsd || hurd || (linux && !android) || netbsd || openbsd || solaris || genode) && cgo
 
 // Package pty is a simple pseudo-terminal package for Unix systems,
 // implemented by calling C functions via cgo.
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/os/signal/signal_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/os/signal/signal_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/os/signal/signal_unix.go	2023-06-01 13:13:31.266417929 +0300
+++ gcc/libgo/go/os/signal/signal_unix.go	2023-06-01 13:57:17.525140308 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows || genode
 
 package signal
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/os/stat_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/os/stat_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/os/stat_unix.go	2023-06-01 13:13:31.254417768 +0300
+++ gcc/libgo/go/os/stat_unix.go	2023-06-01 13:53:39.039241289 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || genode
 
 package os
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/os/sticky_bsd.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/os/sticky_bsd.go
--- contrib/gcc.libgo_etalon/libgo/go/os/sticky_bsd.go	2023-06-01 13:13:31.257417808 +0300
+++ gcc/libgo/go/os/sticky_bsd.go	2023-06-01 13:54:30.206920146 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || (js && wasm) || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || (js && wasm) || netbsd || openbsd || solaris || genode
 
 package os
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/os/sticky_notbsd.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/os/sticky_notbsd.go
--- contrib/gcc.libgo_etalon/libgo/go/os/sticky_notbsd.go	2023-06-01 13:13:31.263417889 +0300
+++ gcc/libgo/go/os/sticky_notbsd.go	2023-06-01 13:55:16.636536180 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build !aix && !darwin && !dragonfly && !freebsd && (!js || !wasm) && !netbsd && !openbsd && !solaris
+//go:build !aix && !darwin && !dragonfly && !freebsd && (!js || !wasm) && !netbsd && !openbsd && !solaris && !genode
 
 package os
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/os/sys_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/os/sys_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/os/sys_unix.go	2023-06-01 13:13:31.263417889 +0300
+++ gcc/libgo/go/os/sys_unix.go	2023-06-01 13:55:48.512959140 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || genode
 
 package os
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/os/user/listgroups_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/os/user/listgroups_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/os/user/listgroups_unix.go	2023-06-01 13:13:31.255417782 +0300
+++ gcc/libgo/go/os/user/listgroups_unix.go	2023-06-01 14:00:25.667637321 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build ((darwin || dragonfly || freebsd || hurd || (js && wasm) || (!android && linux) || netbsd || openbsd || solaris) && (!cgo || osusergo)) || aix || illumos
+//go:build ((darwin || dragonfly || freebsd || hurd || (js && wasm) || (!android && linux) || netbsd || openbsd || solaris) && (!cgo || osusergo)) || aix || illumos || genode
 
 package user
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/os/user/lookup_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/os/user/lookup_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/os/user/lookup_unix.go	2023-06-01 13:13:31.255417782 +0300
+++ gcc/libgo/go/os/user/lookup_unix.go	2023-06-01 14:01:05.991172563 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build (aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || (!android && linux) || netbsd || openbsd || solaris) && (!cgo || osusergo)
+//go:build (aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || (!android && linux) || netbsd || openbsd || solaris || genode) && (!cgo || osusergo)
 
 package user
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/os/wait_unimp.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/os/wait_unimp.go
--- contrib/gcc.libgo_etalon/libgo/go/os/wait_unimp.go	2023-06-01 13:13:31.263417889 +0300
+++ gcc/libgo/go/os/wait_unimp.go	2023-06-01 13:56:16.328328229 +0300
@@ -6,7 +6,7 @@
 // waitid/wait6. netbsd implements wait6, but that is causing test
 // failures, see issue #48789.
 
-//go:build aix || darwin || hurd || (js && wasm) || netbsd || openbsd || solaris
+//go:build aix || darwin || hurd || (js && wasm) || netbsd || openbsd || solaris || genode
 
 package os
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/path/filepath/path_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/path/filepath/path_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/path/filepath/path_unix.go	2023-06-01 13:13:31.114415896 +0300
+++ gcc/libgo/go/path/filepath/path_unix.go	2023-06-01 14:01:37.404589552 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || genode
 
 package filepath
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/runtime/auxv_none.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/runtime/auxv_none.go
--- contrib/gcc.libgo_etalon/libgo/go/runtime/auxv_none.go	2023-06-01 13:13:31.169416631 +0300
+++ gcc/libgo/go/runtime/auxv_none.go	2023-06-01 14:02:20.288158821 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build !linux && !darwin && !dragonfly && !freebsd && !netbsd && !solaris
+//go:build !linux && !darwin && !dragonfly && !freebsd && !netbsd && !solaris && !genode
 
 package runtime
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/runtime/env_posix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/runtime/env_posix.go
--- contrib/gcc.libgo_etalon/libgo/go/runtime/env_posix.go	2023-06-01 13:13:31.157416471 +0300
+++ gcc/libgo/go/runtime/env_posix.go	2023-06-01 14:02:40.534427596 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows || plan9
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows || plan9 || genode
 
 package runtime
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/runtime/lock_sema.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/runtime/lock_sema.go
--- contrib/gcc.libgo_etalon/libgo/go/runtime/lock_sema.go	2023-06-01 13:13:31.162416538 +0300
+++ gcc/libgo/go/runtime/lock_sema.go	2023-06-01 14:03:04.325743439 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || hurd || netbsd || openbsd || plan9 || solaris || windows
+//go:build aix || darwin || hurd || netbsd || openbsd || plan9 || solaris || windows || genode
 
 package runtime
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/runtime/nbpipe_pipe2.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/runtime/nbpipe_pipe2.go
--- contrib/gcc.libgo_etalon/libgo/go/runtime/nbpipe_pipe2.go	2023-06-01 13:13:31.159416498 +0300
+++ gcc/libgo/go/runtime/nbpipe_pipe2.go	2023-06-01 14:03:27.821055360 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris
+//go:build dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || genode
 
 package runtime
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/runtime/netpoll.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/runtime/netpoll.go
--- contrib/gcc.libgo_etalon/libgo/go/runtime/netpoll.go	2023-06-01 13:13:31.162416538 +0300
+++ gcc/libgo/go/runtime/netpoll.go	2023-06-01 14:04:59.357270663 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || windows || genode
 
 package runtime
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/syscall/dirent.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/syscall/dirent.go
--- contrib/gcc.libgo_etalon/libgo/go/syscall/dirent.go	2023-06-01 13:13:31.116415923 +0300
+++ gcc/libgo/go/syscall/dirent.go	2023-06-01 14:06:20.076342448 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || genode
 
 package syscall
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/syscall/env_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/syscall/env_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/syscall/env_unix.go	2023-06-01 13:13:31.116415923 +0300
+++ gcc/libgo/go/syscall/env_unix.go	2023-06-01 14:06:44.446666055 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || plan9
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || plan9 || genode
 
 // Unix environment variables.
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/syscall/forkpipe.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/syscall/forkpipe.go
--- contrib/gcc.libgo_etalon/libgo/go/syscall/forkpipe.go	2023-06-01 13:13:31.122416003 +0300
+++ gcc/libgo/go/syscall/forkpipe.go	2023-06-01 14:11:12.041219894 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || solaris
+//go:build aix || darwin || solaris || genode
 
 package syscall
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/syscall/sockcmsg_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/syscall/sockcmsg_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/syscall/sockcmsg_unix.go	2023-06-01 13:13:31.117415936 +0300
+++ gcc/libgo/go/syscall/sockcmsg_unix.go	2023-06-01 14:12:33.823306204 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || genode
 
 // Socket control messages
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/syscall/sockcmsg_unix_other.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/syscall/sockcmsg_unix_other.go
--- contrib/gcc.libgo_etalon/libgo/go/syscall/sockcmsg_unix_other.go	2023-06-01 13:13:31.119415963 +0300
+++ gcc/libgo/go/syscall/sockcmsg_unix_other.go	2023-06-01 14:12:15.004056221 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || freebsd || hurd || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || freebsd || hurd || linux || netbsd || openbsd || solaris || genode
 
 package syscall
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/syscall/socket_bsd.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/syscall/socket_bsd.go
--- contrib/gcc.libgo_etalon/libgo/go/syscall/socket_bsd.go	2023-06-01 13:13:31.115415909 +0300
+++ gcc/libgo/go/syscall/socket_bsd.go	2023-06-01 14:12:52.439553496 +0300
@@ -4,7 +4,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd openbsd netbsd
+// +build darwin dragonfly freebsd openbsd netbsd genode
 
 package syscall
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/syscall/syscall_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/syscall/syscall_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/syscall/syscall_unix.go	2023-06-01 13:13:31.117415936 +0300
+++ gcc/libgo/go/syscall/syscall_unix.go	2023-06-01 14:13:35.672127799 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || genode
 
 package syscall
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/syscall/timestruct.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/syscall/timestruct.go
--- contrib/gcc.libgo_etalon/libgo/go/syscall/timestruct.go	2023-06-01 13:13:31.119415963 +0300
+++ gcc/libgo/go/syscall/timestruct.go	2023-06-01 14:14:20.190719209 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || genode
 
 package syscall
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/time/sys_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/time/sys_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/time/sys_unix.go	2023-06-01 13:13:31.236417528 +0300
+++ gcc/libgo/go/time/sys_unix.go	2023-06-01 14:15:07.525348056 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || (js && wasm) || linux || netbsd || openbsd || solaris || genode
 
 package time
 
diff -NaurwB contrib/gcc.libgo_etalon/libgo/go/time/zoneinfo_unix.go contrib/libgo-263ef8ff7b07e22b6c5459a220956fdda014ddd9/src/lib/gcc/libgo/go/time/zoneinfo_unix.go
--- contrib/gcc.libgo_etalon/libgo/go/time/zoneinfo_unix.go	2023-06-01 13:13:31.236417528 +0300
+++ gcc/libgo/go/time/zoneinfo_unix.go	2023-06-01 14:15:25.068581128 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || (darwin && !ios) || dragonfly || freebsd || hurd || (linux && !android) || netbsd || openbsd || solaris
+//go:build aix || (darwin && !ios) || dragonfly || freebsd || hurd || (linux && !android) || netbsd || openbsd || solaris || genode
 
 // Parse "zoneinfo" time zone file.
 // This is a fairly standard file format used on OS X, Linux, BSD, Sun, and others.

diff -uwBrN gcc.libgo_etalon2/libgo/go/internal/syscall/unix/net.go libgo-27ecc7632f6a6013fc730a299f98e8d219ac459e/src/lib/gcc/libgo/go/internal/syscall/unix/net.go
--- gcc.libgo_etalon2/libgo/go/internal/syscall/unix/net.go	2023-06-01 20:17:50.680190061 +0300
+++ gcc/libgo/go/internal/syscall/unix/net.go	2023-06-01 23:39:12.209294289 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || genode
 
 package unix
 
diff -uwBrN gcc.libgo_etalon2/libgo/go/net/cgo_resnew.go libgo-27ecc7632f6a6013fc730a299f98e8d219ac459e/src/lib/gcc/libgo/go/net/cgo_resnew.go
--- gcc.libgo_etalon2/libgo/go/net/cgo_resnew.go	2023-06-01 20:17:51.123195996 +0300
+++ gcc/libgo/go/net/cgo_resnew.go	2023-06-02 00:04:53.226484633 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build cgo && !netgo && (aix || darwin || hurd || (linux && !android) || netbsd || solaris)
+//go:build cgo && !netgo && (aix || darwin || hurd || (linux && !android) || netbsd || solaris || genode)
 
 package net
 
diff -uwBrN gcc.libgo_etalon2/libgo/go/net/cgo_sockold.go libgo-27ecc7632f6a6013fc730a299f98e8d219ac459e/src/lib/gcc/libgo/go/net/cgo_sockold.go
--- gcc.libgo_etalon2/libgo/go/net/cgo_sockold.go	2023-06-01 20:17:51.135196157 +0300
+++ gcc/libgo/go/net/cgo_sockold.go	2023-06-02 00:05:43.103170768 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build cgo && !netgo && (aix || darwin || dragonfly || freebsd || hurd || netbsd || openbsd)
+//go:build cgo && !netgo && (aix || darwin || dragonfly || freebsd || hurd || netbsd || openbsd || genode)
 
 package net
 
diff -uwBrN gcc.libgo_etalon2/libgo/go/net/cgo_unix.go libgo-27ecc7632f6a6013fc730a299f98e8d219ac459e/src/lib/gcc/libgo/go/net/cgo_unix.go
--- gcc.libgo_etalon2/libgo/go/net/cgo_unix.go	2023-06-01 20:17:51.074195339 +0300
+++ gcc/libgo/go/net/cgo_unix.go	2023-06-02 00:01:39.331808417 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build cgo && !netgo && (aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris)
+//go:build cgo && !netgo && (aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || genode)
 
 package net
 
diff -uwBrN gcc.libgo_etalon2/libgo/go/net/fd_posix.go libgo-27ecc7632f6a6013fc730a299f98e8d219ac459e/src/lib/gcc/libgo/go/net/fd_posix.go
--- gcc.libgo_etalon2/libgo/go/net/fd_posix.go	2023-06-01 20:17:51.078195393 +0300
+++ gcc/libgo/go/net/fd_posix.go	2023-06-01 23:55:50.599990651 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || windows
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || windows || genode
 
 package net
 
diff -uwBrN gcc.libgo_etalon2/libgo/go/net/unixsock_readmsg_cloexec.go libgo-27ecc7632f6a6013fc730a299f98e8d219ac459e/src/lib/gcc/libgo/go/net/unixsock_readmsg_cloexec.go
--- gcc.libgo_etalon2/libgo/go/net/unixsock_readmsg_cloexec.go	2023-06-01 20:17:51.128196063 +0300
+++ gcc/libgo/go/net/unixsock_readmsg_cloexec.go	2023-06-02 00:08:14.721254550 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || freebsd || hurd || solaris
+//go:build aix || darwin || freebsd || hurd || solaris || genode
 
 package net
 
diff -uwBrN gcc.libgo_etalon2/libgo/go/os/exec/internal/fdtest/exists_unix.go libgo-27ecc7632f6a6013fc730a299f98e8d219ac459e/src/lib/gcc/libgo/go/os/exec/internal/fdtest/exists_unix.go
--- gcc.libgo_etalon2/libgo/go/os/exec/internal/fdtest/exists_unix.go	2023-06-01 20:17:50.953193718 +0300
+++ gcc/libgo/go/os/exec/internal/fdtest/exists_unix.go	2023-06-01 20:18:01.383333449 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris
+//go:build aix || darwin || dragonfly || freebsd || hurd || linux || netbsd || openbsd || solaris || genode
 
 // Package fdtest provides test helpers for working with file descriptors across exec.
 package fdtest
diff -uwBrN gcc.libgo_etalon2/libgo/go/os/stat_atim.go libgo-27ecc7632f6a6013fc730a299f98e8d219ac459e/src/lib/gcc/libgo/go/os/stat_atim.go
--- gcc.libgo_etalon2/libgo/go/os/stat_atim.go	2023-06-01 20:17:50.948193651 +0300
+++ gcc/libgo/go/os/stat_atim.go	2023-06-01 23:44:15.383419117 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build hurd linux openbsd solaristag
+// +build hurd linux openbsd solaristag genode
 
 package os
 
diff -uwBrN gcc.libgo_etalon2/libgo/go/os/sys_uname.go libgo-27ecc7632f6a6013fc730a299f98e8d219ac459e/src/lib/gcc/libgo/go/os/sys_uname.go
--- gcc.libgo_etalon2/libgo/go/os/sys_uname.go	2023-06-01 20:17:50.958193785 +0300
+++ gcc/libgo/go/os/sys_uname.go	2023-06-01 23:47:35.602141882 +0300
@@ -4,7 +4,7 @@
 
 // For systems which only store the hostname in uname (Solaris).
 
-// +build aix hurd solaris irix rtems
+// +build aix hurd solaris irix rtems genode
 
 package os
 
diff -uwBrN gcc.libgo_etalon2/libgo/go/os/user/cgo_lookup_unix.go libgo-27ecc7632f6a6013fc730a299f98e8d219ac459e/src/lib/gcc/libgo/go/os/user/cgo_lookup_unix.go
--- gcc.libgo_etalon2/libgo/go/os/user/cgo_lookup_unix.go	2023-06-01 20:17:50.946193625 +0300
+++ gcc/libgo/go/os/user/cgo_lookup_unix.go	2023-06-01 23:54:28.062849517 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-//go:build (aix || darwin || dragonfly || freebsd || hurd || (!android && linux) || netbsd || openbsd || solaris) && cgo && !osusergo
+//go:build (aix || darwin || dragonfly || freebsd || hurd || (!android && linux) || netbsd || openbsd || solaris || genode) && cgo && !osusergo
 
 package user
 
diff -uwBrN gcc.libgo_etalon2/libgo/go/os/user/decls_unix.go libgo-27ecc7632f6a6013fc730a299f98e8d219ac459e/src/lib/gcc/libgo/go/os/user/decls_unix.go
--- gcc.libgo_etalon2/libgo/go/os/user/decls_unix.go	2023-06-01 20:17:50.946193625 +0300
+++ gcc/libgo/go/os/user/decls_unix.go	2023-06-01 23:53:37.873155438 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build darwin dragonfly freebsd hurd linux netbsd openbsd
+// +build darwin dragonfly freebsd hurd linux netbsd openbsd genode
 // +build cgo
 
 package user
diff -uwBrN gcc.libgo_etalon2/libgo/go/syscall/libcall_posix_utimesnano.go libgo-27ecc7632f6a6013fc730a299f98e8d219ac459e/src/lib/gcc/libgo/go/syscall/libcall_posix_utimesnano.go
--- gcc.libgo_etalon2/libgo/go/syscall/libcall_posix_utimesnano.go	2023-06-01 20:17:50.655189726 +0300
+++ gcc/libgo/go/syscall/libcall_posix_utimesnano.go	2023-06-01 23:41:49.163430030 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build aix darwin dragonfly freebsd hurd openbsd netbsd solaris
+// +build aix darwin dragonfly freebsd hurd openbsd netbsd solaris genode
 
 // General POSIX version of UtimesNano.
 
diff -uwBrN gcc.libgo_etalon2/libgo/go/syscall/syscall_funcs_stubs.go libgo-27ecc7632f6a6013fc730a299f98e8d219ac459e/src/lib/gcc/libgo/go/syscall/syscall_funcs_stubs.go
--- gcc.libgo_etalon2/libgo/go/syscall/syscall_funcs_stubs.go	2023-06-01 20:17:50.655189726 +0300
+++ gcc/libgo/go/syscall/syscall_funcs_stubs.go	2023-06-01 21:57:03.022191957 +0300
@@ -2,7 +2,7 @@
 // Use of this source code is governed by a BSD-style
 // license that can be found in the LICENSE file.
 
-// +build aix rtems
+// +build aix rtems genode
 
 // These are stubs.
 
