diff --git a/include/c.h b/include/c.h
index cbc99669f..32fbb1bc3 100644
--- a/include/c.h
+++ b/include/c.h
@@ -390,6 +390,7 @@ fail:
 
 static inline int xusleep(useconds_t usec)
 {
+/*
 #ifdef HAVE_NANOSLEEP
 	struct timespec waittime = {
 		.tv_sec   =  usec / 1000000L,
@@ -397,10 +398,13 @@ static inline int xusleep(useconds_t usec)
 	};
 	return nanosleep(&waittime, NULL);
 #elif defined(HAVE_USLEEP)
+*/
 	return usleep(usec);
+/*
 #else
 # error	"System with usleep() or nanosleep() required!"
 #endif
+*/
 }
 
 /*
