diff --git a/src/google/protobuf/stubs/mutex.h b/src/google/protobuf/stubs/mutex.h
--- a/src/google/protobuf/stubs/mutex.h
+++ b/src/google/protobuf/stubs/mutex.h
@@ -30,7 +30,7 @@
 #ifndef GOOGLE_PROTOBUF_STUBS_MUTEX_H_
 #define GOOGLE_PROTOBUF_STUBS_MUTEX_H_
 
-#include <mutex>
+#include <base/mutex.h>
 
 #ifdef GOOGLE_PROTOBUF_SUPPORT_WINDOWS_XP
 
@@ -96,15 +96,15 @@ class PROTOBUF_EXPORT CriticalSectionLock {
 class GOOGLE_PROTOBUF_CAPABILITY("mutex") PROTOBUF_EXPORT WrappedMutex {
  public:
   WrappedMutex() = default;
-  void Lock() GOOGLE_PROTOBUF_ACQUIRE() { mu_.lock(); }
-  void Unlock() GOOGLE_PROTOBUF_RELEASE() { mu_.unlock(); }
+  void Lock() GOOGLE_PROTOBUF_ACQUIRE() { mu_.acquire(); }
+  void Unlock() GOOGLE_PROTOBUF_RELEASE() { mu_.release(); }
   // Crash if this Mutex is not held exclusively by this thread.
   // May fail to crash when it should; will never crash when it should not.
   void AssertHeld() const {}
 
  private:
 #ifndef GOOGLE_PROTOBUF_SUPPORT_WINDOWS_XP
-  std::mutex mu_;
+  Genode::Mutex mu_;
 #else  // ifndef GOOGLE_PROTOBUF_SUPPORT_WINDOWS_XP
   CriticalSectionLock mu_;
 #endif  // #ifndef GOOGLE_PROTOBUF_SUPPORT_WINDOWS_XP
