commit 420f5bb78bfd61825ad5d3bc52a27469f298d17b
Author: Pirmin Duss <pirmin.duss@gapfruit.com>
Date:   Mon Sep 23 16:33:08 2019 +0200

    run_once fix

diff --git a/src/google/protobuf/stubs/once.h b/src/google/protobuf/stubs/once.h
index 070d36d..3602ee7 100644
--- a/src/google/protobuf/stubs/once.h
+++ b/src/google/protobuf/stubs/once.h
@@ -31,19 +31,16 @@
 #ifndef GOOGLE_PROTOBUF_STUBS_ONCE_H__
 #define GOOGLE_PROTOBUF_STUBS_ONCE_H__
 
-#include <mutex>
-#include <utility>
-
 #include <google/protobuf/port_def.inc>
 
 namespace google {
 namespace protobuf {
 namespace internal {
 
-using once_flag = std::once_flag;
-template <typename... Args>
-void call_once(Args&&... args ) {
-  std::call_once(std::forward<Args>(args)...);
+using once_flag = int;
+template <typename FN, typename... Args>
+void call_once(once_flag, FN fn, Args&&... args ) {
+  fn(std::forward<Args>(args)...);
 }
 
 }  // namespace internal
