--- a/src/main.c	2017-04-29 12:48:50.000000000 +0200
+++ b/src/main.c	2018-09-25 23:02:45.496486188 +0200
@@ -248,7 +248,7 @@
 
 int main(int argc, char **argv)
 {
-	bool_t nodaemon = false;
+	bool_t nodaemon = true;
 #ifdef POSIX_PRIORITY_SCHEDULING
 	bool_t realtime = false;
 #endif
--- a/src/server.c	2017-04-29 12:48:50.000000000 +0200
+++ b/src/server.c	2018-09-25 23:12:37.492917577 +0200
@@ -71,9 +71,10 @@
 	hasv4 = (errno == EAFNOSUPPORT || errno == EPROTONOSUPPORT) ? false : true;
 	if (!(testsocket < 0)) close(testsocket);
 
-	testsocket = socket(PF_INET6, SOCK_STREAM, 0);
-	hasv6 = (errno == EAFNOSUPPORT || errno == EPROTONOSUPPORT) ? false : true;
-	if (!(testsocket < 0)) close(testsocket);
+	// testsocket = socket(PF_INET6, SOCK_STREAM, 0);
+	// hasv6 = (errno == EAFNOSUPPORT || errno == EPROTONOSUPPORT) ? false : true;
+	// if (!(testsocket < 0)) close(testsocket);
+	hasv6 = false;
 
 	if(!hasv4)
 	{
