diff -ru contrib/libatomic-cb6e6b96c6eb2789f5ae26549443d3d58185e4ef/src/lib/gcc/config.sub inno/config.sub
--- contrib/libatomic-cb6e6b96c6eb2789f5ae26549443d3d58185e4ef/src/lib/gcc/config.sub	2018-01-03 07:25:18.000000000 +0300
+++ inno/config.sub	2020-06-22 16:24:54.000000000 +0300
@@ -313,6 +313,10 @@
 				basic_machine=m68k-ericsson
 				os=ose
 				;;
+			genode)
+				basic_machine=x86_64-pc
+				os=-elf
+				;;
 			gmicro)
 				basic_machine=tron-gmicro
 				os=sysv
diff -ru contrib/libatomic-cb6e6b96c6eb2789f5ae26549443d3d58185e4ef/src/lib/gcc/libatomic/configure.tgt inno/libatomic/configure.tgt
--- contrib/libatomic-cb6e6b96c6eb2789f5ae26549443d3d58185e4ef/src/lib/gcc/libatomic/configure.tgt	2018-03-09 15:32:22.000000000 +0300
+++ inno/libatomic/configure.tgt	2020-06-22 16:24:54.000000000 +0300
@@ -137,7 +137,7 @@
   *-*-linux* | *-*-gnu* | *-*-k*bsd*-gnu \
   | *-*-netbsd* | *-*-freebsd* | *-*-openbsd* | *-*-dragonfly* \
   | *-*-solaris2* | *-*-sysv4* | *-*-irix6* | *-*-osf* | *-*-hpux11* \
-  | *-*-darwin* | *-*-aix* | *-*-cygwin*)
+  | *-*-darwin* | *-*-aix* | *-*-cygwin*| x86_64-pc-elf)
	# POSIX system.  The OS is supported.
	config_path="${config_path} posix"
	;;
