diff -ru contrib/libatomic-cb6e6b96c6eb2789f5ae26549443d3d58185e4ef/src/lib/gcc/libatomic/configure.tgt genode/libatomic/configure.tgt
--- contrib/libatomic-cb6e6b96c6eb2789f5ae26549443d3d58185e4ef/src/lib/gcc/libatomic/configure.tgt	2018-03-09 15:32:22.000000000 +0300
+++ genode/libatomic/configure.tgt	2020-06-22 16:24:54.000000000 +0300
@@ -137,7 +137,7 @@
   *-*-linux* | *-*-gnu* | *-*-k*bsd*-gnu \
   | *-*-netbsd* | *-*-freebsd* | *-*-openbsd* | *-*-dragonfly* \
   | *-*-solaris2* | *-*-sysv4* | *-*-irix6* | *-*-osf* | *-*-hpux11* \
-  | *-*-darwin* | *-*-aix* | *-*-cygwin*)
+  | *-*-darwin* | *-*-aix* | *-*-cygwin* | x86_64-pc-elf | aarch64-none-elf)
	# POSIX system.  The OS is supported.
	config_path="${config_path} posix"
	;;
