commit 375d5368e19718d37c8d131690e00b495a704af2
Author: Sebastian Sumpf <sebastian.sumpf@genode-labs.com>
Date:   Mon Sep 10 19:11:29 2018 +0200

    patch: no support for khelper

diff --git a/hotspot/src/cpu/arm/vm/vm_version_arm_32.cpp b/hotspot/src/cpu/arm/vm/vm_version_arm_32.cpp
index 10a2408..b35d68f 100644
--- a/hotspot/src/cpu/arm/vm/vm_version_arm_32.cpp
+++ b/hotspot/src/cpu/arm/vm/vm_version_arm_32.cpp
@@ -217,7 +217,11 @@ void VM_Version::initialize() {
 
   get_os_cpu_info();
 
+#ifdef __GENODE__
+  _kuser_helper_version = 0;
+#else
   _kuser_helper_version = *(int*)KUSER_HELPER_VERSION_ADDR;
+#endif
 
 #ifdef COMPILER2
   // C2 is only supported on v7+ VFP at this time

diff --git a/hotspot/src/cpu/arm/vm/assembler_arm_32.hpp b/hotspot/src/cpu/arm/vm/assembler_arm_32.hpp
index e32f6a9..2f1411e 100644
--- a/hotspot/src/cpu/arm/vm/assembler_arm_32.hpp
+++ b/hotspot/src/cpu/arm/vm/assembler_arm_32.hpp
@@ -1234,10 +1234,17 @@ extern int  __aeabi_dcmpgt(double, double);

 // Imported code from glibc soft-fp bundle for
 // calculation accuracy improvement. See CR 6757269.
+
+/*
 extern double __aeabi_fadd_glibc(float, float);
 extern double __aeabi_fsub_glibc(float, float);
 extern double __aeabi_dadd_glibc(double, double);
 extern double __aeabi_dsub_glibc(double, double);
+*/
+#define __aeabi_dadd_glibc __aeabi_dadd
+#define __aeabi_dsub_glibc __aeabi_dsub
+#define __aeabi_fadd_glibc __aeabi_fadd
+#define __aeabi_fsub_glibc __aeabi_fsub
 };
 #endif // __SOFTFP__
