diff --git a/include/grpc/impl/codegen/sync_custom.h b/include/grpc/impl/codegen/sync_custom.h
--- a/include/grpc/impl/codegen/sync_custom.h
+++ b/include/grpc/impl/codegen/sync_custom.h
@@ -21,17 +21,24 @@
 
 #include <grpc/impl/codegen/port_platform.h>
 
-#include <grpc/impl/codegen/sync_generic.h>
+#ifdef __cplusplus
+}
+#endif
+#include <base/lock.h>
+#ifdef __cplusplus
+extern "C" {
+#include <pthread.h>
+#endif
 
 /* Users defining GPR_CUSTOM_SYNC need to define the following macros. */
 
 #ifdef GPR_CUSTOM_SYNC
 
-typedef GPR_CUSTOM_MU_TYPE gpr_mu;
-typedef GPR_CUSTOM_CV_TYPE gpr_cv;
-typedef GPR_CUSTOM_ONCE_TYPE gpr_once;
+typedef Genode::Lock gpr_mu;
+typedef pthread_cond_t gpr_cv;
+typedef pthread_once_t gpr_once;
 
-#define GPR_ONCE_INIT GPR_CUSTOM_ONCE_INIT
+#define GPR_ONCE_INIT PTHREAD_ONCE_INIT
 
 #endif
 
diff --git a/include/grpcpp/impl/codegen/sync.h b/include/grpcpp/impl/codegen/sync.h
--- a/include/grpcpp/impl/codegen/sync.h
+++ b/include/grpcpp/impl/codegen/sync.h
@@ -25,7 +25,7 @@
 #include <pthread.h>
 #endif
 
-#include <mutex>
+#include <base/mutex.h>
 
 #include <grpc/impl/codegen/log.h>
 #include <grpc/impl/codegen/sync.h>
@@ -58,7 +58,7 @@ class Mutex {
  private:
   union {
     gpr_mu mu_;
-    std::mutex do_not_use_sth_;
+    Genode::Mutex do_not_use_sth_;
 #ifdef GPR_HAS_PTHREAD_H
     pthread_mutex_t do_not_use_pth_;
 #endif
