diff --git a/third_party/abseil-cpp/absl/base/internal/raw_logging.cc b/third_party/abseil-cpp/absl/base/internal/raw_logging.cc
index 40cea55..6707e12 100644
--- a/third_party/abseil-cpp/absl/base/internal/raw_logging.cc
+++ b/third_party/abseil-cpp/absl/base/internal/raw_logging.cc
@@ -41,8 +41,8 @@
 
 #include <unistd.h>
 
-#define ABSL_HAVE_POSIX_WRITE 1
-#define ABSL_LOW_LEVEL_WRITE_SUPPORTED 1
+#undef ABSL_HAVE_POSIX_WRITE
+#undef ABSL_LOW_LEVEL_WRITE_SUPPORTED
 #else
 #undef ABSL_HAVE_POSIX_WRITE
 #endif
@@ -52,8 +52,8 @@
 // for low level operations that want to avoid libc.
 #if (defined(__linux__) || defined(__FreeBSD__)) && !defined(__ANDROID__)
 #include <sys/syscall.h>
-#define ABSL_HAVE_SYSCALL_WRITE 1
-#define ABSL_LOW_LEVEL_WRITE_SUPPORTED 1
+#undef ABSL_HAVE_SYSCALL_WRITE
+#define ABSL_LOW_LEVEL_WRITE_SUPPORTED 0
 #else
 #undef ABSL_HAVE_SYSCALL_WRITE
 #endif
@@ -61,8 +61,8 @@
 #ifdef _WIN32
 #include <io.h>
 
-#define ABSL_HAVE_RAW_IO 1
-#define ABSL_LOW_LEVEL_WRITE_SUPPORTED 1
+#undef ABSL_HAVE_RAW_IO
+#undef ABSL_LOW_LEVEL_WRITE_SUPPORTED
 #else
 #undef ABSL_HAVE_RAW_IO
 #endif
